src/main/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -327,16 +330,19 @@ private PhysicalLocation keeperOrNull(final List<? extends PhysicalLocation> lis
     /** Simple method to test whether two physical locations are close enough to each other to be deemed optical dupes. */
     private boolean closeEnough(final PhysicalLocation lhs, final PhysicalLocation rhs, final int distance) {
         return lhs != rhs &&                                    // no comparing an object to itself (checked using object identity)!
-               lhs.hasLocation() && rhs.hasLocation() &&        // no comparing objects without locations
-               lhs.getReadGroup() == rhs.getReadGroup() &&      // must be in the same RG to be optical duplicates
-               lhs.getTile()      == rhs.getTile()      &&      // and the same tile
-               Math.abs(lhs.getX() - rhs.getX()) <= distance &&
-               Math.abs(lhs.getY() - rhs.getY()) <= distance;
+                lhs.hasLocation() && rhs.hasLocation() &&        // no comparing objects without locations
+                lhs.getReadGroup() == rhs.getReadGroup() &&      // must be in the same RG to be optical duplicates
+                lhs.getTile() == rhs.getTile() &&      // and the same tile
+                closeEnoughShort(lhs, rhs, distance);
     }
 
     private boolean closeEnoughShort(final PhysicalLocation lhs, final PhysicalLocation rhs, final int distance) {
         return lhs != rhs &&
-                Math.abs(lhs.getX() - rhs.getX()) <= distance &&
-                Math.abs(lhs.getY() - rhs.getY()) <= distance;
+                // Since the X and Y coordinates are constrained to short values, they can also become negative when
```

Comment:
I had to do some arithmetic to convince myself that this works. Assuming your goal is to make it so that both 32768 and 32767 adjacent without disturbing the  65536 and 65535 adjacency I think this succeeded. However I do not think it is necessary to perform the operation forwards and backwards, the absolute value operation (even accounting for overflowing from the shorts) will not be changed by flipping the order by virtue of the fact that this calculation is being performed in integer space whereas the underlying data overflowed in short space. Unless your scaling factor is in the region of `Short.MAX_VALUE/2` you should never have an overflow here thus you should be able to get away without flipping the comparison.